README.md

```diff
@@ -100,17 +102,89 @@ Below are the steps for running and performing data analysis on your local machi
+
+##### To run this job on a local machine:
+```shell
+poetry run python -m pipeline.main
+```
+
+##### To run this job from Azure:
+Use Azure Data Factory to execute the python file `dbfs:/pipeline/main.py` on databricks.
+
+
+#### 2. Run forecast and output to blob
+
+Related files: [`pipeline/jobs/run_forecast_and_output_to_blob.py`]
+
+This job generate spend forecast for the future 12 months base on spend data.
 
-#### How to run data pipeline
+Currently it works with the forecast model ARMA / ARIMA and SARIMA.
+It pulls data from the blob storage, compares different models performance for each Category/MarketSector combination, then for each combination use the better performed model to generate forecast, finally output the forecast to blob storage as .csv file.
+
+Note: For the moment, this job only run forecast for several pre-defined combinations, due to the long computation time needed for running SARIMA model (around 10 minutes per each combination).
+This behaviour can be changed easily in the related python file
+
+This jobs requires the `SpendDataFilledMissingMonth` data file in blob storage, which is generated by the job above.
+The output is stored on blob storage in folder `ForecastOutput`.
+
+##### To run this job on a local machine:
+```shell
+poetry run python -m pipeline.jobs.run_forecast_and_output_to_blob
+```
+
+##### To run this job from Azure:
+Use Azure Data Factory to execute the python file `dbfs:/pipeline/jobs/run_forecast_and_output_to_blob.py` on databricks.
+
+
+
+#### 3. Get active combinations
+
+Related files: [`pipeline/jobs/get_active_combinations.py`]
+
+This job generate a list of active Category / MarketSector combinations base on spend data and Framework status.
+
+It finds all frameworks of Status `Live` or `Expired - Data Still Received`, then from the spend data find all Category / MarketSector combinations that are related to such frameworks. Finally it returns a list of Category / MarketSector combination.
+
+
+This jobs requires the 
+`Customers`, `FrameworkCategoryPillar`, `SpendAggregated` data file in blob storage, which are pulled from the tables under the same name in SQL Server.
+There is a pipeline job in Azure Data Factory to pull those file from SQL server to blob storage.
+The output is stored on blob storage in folder `ActiveCombinations`.
+
+##### To run this job on a local machine:
+```shell
+poetry run python -m pipeline.jobs.get_active_combinations
+```
 
-Below are the steps to run the data pipeline
+##### To run this job from Azure:
+Use Azure Data Factory to execute the python file `dbfs:/pipeline/jobs/get_active_combinations.py` on databricks.
 
-1. TODO: Write about pipeline
+-------------------------------------------------------
 
 ### Testing
 The unit test of this repo is located in `test` directory. You can use the below command in a terminal to run the full \
 test suite:
 
 ```shell
-python3 -m unittest discover -p '*_test.py' -s test
+pytest .
```

Comment:
Think this would need to be `poetry run pyest` wouldn't it?

Or even better -  `make unit_testing`

Reply:
Thanks! Updated this to `make unit_testing`